types/prop-types/index.d.ts

```diff
@@ -0,0 +1,35 @@
+// Type definitions for prop-types v15.5.4
+// Project: https://github.com/reactjs/prop-types
+// Definitions by: DovydasNavickas <https://github.com/DovydasNavickas>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare module PropTypes {
+    type Validator<T> = (object: T, key: string, componentName: string, ...rest: any[]) => Error | null;
+
+    interface Requireable<T> extends Validator<T> {
+        isRequired: Validator<T>;
+    }
+
+    type ValidationMap<T> = {[K in keyof T]?: Validator<T> };
+
+    let any: Requireable<any>;
+    let array: Requireable<any>;
+    let bool: Requireable<any>;
```

Comment:
Should this be `Requireable<boolean>`?
Also, prefer `const` if this is not intended to be modified.